### PR TITLE
Prefer explicit passing of CChainParams to the Params() global.

### PR DESCRIPTION
**Pull request by nuttycom**
This is (intended to be) a pure refactoring, no functional changes included.

### CODE REVIEW

File: src/gtest/test_mempool.cpp [hunk 1]
```diff
@@ -166,7 +166,7 @@ TEST(Mempool, SproutV3TxWhenOverwinterActive) {
     CTransaction tx1(mtx);
 
     LOCK(cs_main);
-    EXPECT_FALSE(AcceptToMemoryPool(pool, state1, tx1, false, &missingInputs));
+    EXPECT_FALSE(AcceptToMemoryPool(Params(),pool, state1, tx1, false, &missingInputs));
```
Comment (daira) [inline src/gtest/test_mempool.cpp:169]:
```suggestion
    EXPECT_FALSE(AcceptToMemoryPool(Params(), pool, state1, tx1, false, &missingInputs));
```

Comment (daira) [reply src/gtest/test_mempool.cpp:169]: And similarly below.

**Comment by nuttycom** [comment]
@zkbot try

**Review by therealyingtong — APPROVED** [review]
utACK

**Review by daira — APPROVED** [review]
utACK

**Comment by daira** [comment]
@zkbot r+

**Pull request merged**